### PR TITLE
Fix flaky tests: test which checks a number of ticks on a view, and log event with reserved properties test

### PR DESCRIPTION
**Pull request by 0xnm**
### What does this PR do?

This PR fixes 2 flaky tests:

* The problem is that if we received tick only once, then it is not enough to calculate a metric `number of ticks on a view`, because we need at least 2 points per this code https://github.com/DataDog/dd-sdk-android/blob/0e9ac1420b70390b58b923902638e57ffc25af1c/dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt#L85-L90

* There is a possibility to have `LogEvent.Usr.additionalProperties` with keys which are reserved and dropped during the serialization, but assertion was missing this drop part. Solved by making `RESERVED_PROPERTIES` as `internal` in generated models and by adding a link to it in the assertion.

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

**Review by mariusc83 — APPROVED** [review]

File: dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/event/LogEventSerializer.kt [hunk 1]
```diff
@@ -25,10 +25,9 @@ internal class LogEventSerializer(
         val sanitizedTags = dataConstraints
             .validateTags(log.ddtags.split(","))
             .joinToString(",")
-        // TODO: 13/04/2021 RUMM-1298 Handle the RESERVED_ATTRIBUTES in the `toJson` method
         val sanitizedAttributes = dataConstraints
             .validateAttributes(log.additionalProperties)
-            .filterKeys { it.isNotBlank() && it !in RESERVED_PROPERTIES }
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/event/LogEventSerializer.kt:31]: nice one, I think you can mark that task as done ;)

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt [hunk 2]
```diff
@@ -3142,7 +3142,12 @@ internal class RumViewScopeTest {
         )
 
         // Then
-        val expectedTotal = cpuTicks.last() - cpuTicks.first()
+        val expectedTotal = if (cpuTicks.size > 1) {
```
Comment (mariusc83) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt:3145]: I have the feeling this should be part from another PR ?

**Review by mariusc83 — CHANGES_REQUESTED** [review]

**Review by mariusc83 — APPROVED** [review]

File: dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/event/LogEventSerializer.kt (hunk 1 above)
Comment (0xnm) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/event/LogEventSerializer.kt:31]: it was already marked as `DONE` by you :)

Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/event/LogEventSerializer.kt:31]: My bad then :(. I guess I forgot to remove that code when I replaced that method. Thanks for taking care of it.

**Review by xgouchet — APPROVED** [review]
